internal/service/certificatesigningrequest.go

```diff
@@ -310,8 +310,19 @@ func (h *ServiceHandler) ReplaceCertificateSigningRequest(ctx context.Context, r
 			approveResp, _ := h.ApproveCertificateSigningRequest(ctx, approveReq)
 			_, ok := approveResp.(server.ApproveCertificateSigningRequest200JSONResponse)
 			if !ok {
-				msg := fmt.Sprintf("enrollment CSR for %s could not be auto-approved: %s", request.Name, approveResp)
-				return server.ReplaceCertificateSigningRequest400JSONResponse{Message: msg}, nil
+				// add failed status if newly created enrollment CSR could not be approved;
+				// do not update status on pre-existing enrollment CSR with same name
+				if created {
+					c := []api.Condition{{
+						Message: "CSR could not be auto-approved",
+						Reason:  "CSR could not be auto-approved",
+						Type:    "Failed"}}
+					err := h.store.CertificateSigningRequest().UpdateConditions(ctx, orgId, request.Name, c)
+					if err != nil {
+						return nil, fmt.Errorf("enrollment CSR for %s could not be auto-approved: %s\nstatus condition could not be set to 'failed': %s", request.Name, approveResp, err)
+					}
```

Comment:
Did you mean to return here in a non-error case?` `If created` is always an error case, but maybe that is expected.

Reply:
That is actually expected, but I'm now noticing another problem - hold on, I will update.

Reply:
Okay, I simplified this a bit because we actually don't need to try auto-approving at all unless it was `created`.